pkg/new_storage/general.go

```diff
@@ -151,7 +163,7 @@ func (bd *BackupDestination) BackupList() ([]Backup, error) {
 		if err := json.Unmarshal(b, &m); err != nil {
 			result = append(result, Backup{
 				metadata.BackupMetadata{
-					BackupName: strings.Trim(o.Name(), "/"),
+					BackupName: metadataFilePath,
```

Comment:
looks like you broke 
`clickhouse-backup list remote`
for other storage types
you should set `backup_name` here
not `backup_name/metadata.json`

Reply:
yes, I will check that. thanks.